spec/controllers/queue_items_controller_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'spec_helper'
+require 'shoulda-matchers'
+
+include Testable
+
+
+describe QueueItemsController do
+  it { should use_before_action :require_login }
+
+  describe 'GET index' do
+    it 'sets @queue_items to the queue items of the logged in user' do
+      user = User.find test_login
+      queue_items = 2.times.map { |n| Fabricate :queue_item, user: user, position: n + 1 }
+      get :index
+      expect(assigns :queue_items).to eq queue_items
+    end
+  end
+
+  describe 'POST create' do
+    let(:user) { User.find test_login }
+    let(:video) { Fabricate :video }
+    let(:video_2) { Fabricate :video }
+
+    before do
+      user
```

Comment:
You don't need this if you change line 20 to use `let!` instead. `let` is lazily evaluated, `let!` is not.